src/data_structure/fieldWritable.java

```diff
@@ -29,6 +29,20 @@ public void setColumnFieldInfo(fieldInfo columnFieldInfo) {
 
     public void setPossibilities(boolean[] possibilities) {
         this.possibilities = possibilities;
+        possibilitiesCount = 0;
+        for (boolean i : possibilities) {
+            if (i) possibilitiesCount++;
+        }
+        if (possibilitiesCount == 1) state = State.FILLED;
+        if (possibilitiesCount == 0) state = State.UNFILLED;//TODO error
+    }
+
+    public void revokePossibility(int i) {
+        if (0 <= i && i <= 9 && possibilities[i - 1] == true) {
+            possibilities[i - 1] = false;
+            possibilitiesCount--;
+        }
+        if (possibilitiesCount == 1) state = State.FILLED;
```

Comment:
Moreover if possibilities count = 1 we should call setField() function to update position. For example we need to update positions around filled field.